File: WindowsRegular Expressions/4975sourcecode/chapter6/6-22/recipe6-22.vb
Imports System
Imports System.IO
Imports System.Text.RegularExpressions
Public Class Recipe

	Private Shared _Regex As Regex = _
		New Regex("^(?<path>[A-Za-z]:\\[^/:*?""<>|]+)\((?<line>\d+),(?<pos>\d+)\):")

	Public Sub Run(ByVal fileName As String)
		Dim line As String
		Dim lineNbr As Integer = 0
		Dim sr As StreamReader = File.OpenText(fileName)
		line = sr.ReadLine
		While Not line Is Nothing
			If _Regex.IsMatch(line) = True Then
				Console.WriteLine("Error in file:  '{0}' at line '{1}', position '{2}'", _
						_Regex.Match(line).Result("${path}"), _
						_Regex.Match(line).Result("${line}"), _
						_Regex.Match(line).Result("${pos}"))
			End If
			line = sr.ReadLine
		End While
		sr.Close()
	End Sub

	Public Shared Sub Main(ByVal args As String())
		Dim r As Recipe = New Recipe
		r.Run(args(0))
	End Sub

End Class

File: WindowsRegular Expressions/4975sourcecode/chapter6/6-22/recipe6-22.cs
using System;
using System.IO;
using System.Text.RegularExpressions;

public class Recipe
{
	private static Regex _Regex = new Regex( @"^(?<path>[A-Za-z]:\\[^/:*?""<>|]+)\((?<line>\d+),(?<pos>\d+)\):" );
	
	public void Run(string fileName)
	{
		String line;
		using (StreamReader sr = new StreamReader(fileName))
		{
			while(null != (line = sr.ReadLine()))
			{
				if (_Regex.IsMatch(line))
				{
					Console.WriteLine("Error in file:  '{0}' at line '{1}', position '{2}'", 
						_Regex.Match(line).Result("${path}"),
						_Regex.Match(line).Result("${line}"),
						_Regex.Match(line).Result("${pos}"));
				}
			}

		}
	}	

	public static void Main( string[] args )
	{
		Recipe r = new Recipe();
		r.Run(args[0]);
	}
}
